src/server.rs

```diff
@@ -146,6 +146,33 @@ async fn serve(
     }
 }
 
+/// Remove a file from the upload directory.
+#[delete("/{file}")]
+async fn delete(
+    request: HttpRequest,
+    file: web::Path<String>,
+    config: web::Data<RwLock<Config>>,
+) -> Result<HttpResponse, Error> {
+    let config = config
+        .read()
+        .map_err(|_| error::ErrorInternalServerError("cannot acquire config"))?;
+    let path = config.server.upload_path.join(&*file);
+    let path = util::glob_match_file(path)?;
+    let connection = request.connection_info().clone();
+    let host = connection.realip_remote_addr().unwrap_or("unknown host");
+    let tokens = config.get_delete_tokens();
+    if tokens.is_none() {
+        log::warn!("delete endpoint not served because there are no delete_tokens set");
+        return Err(error::ErrorForbidden("endpoint is not exposed\n"));
+    }
+    auth::check(host, request.headers(), tokens)?;
+    if !path.is_file() || !path.exists() {
+        return Err(error::ErrorNotFound("file is not found or expired :(\n"));
+    }
+    fs::remove_file(path).expect("Could not delete file");
```

Comment:
Can we use `error!` here to log the error and return 500?

Reply:
done